bin/usgscsm_cam_test.cc

```diff
@@ -1,74 +1,211 @@
+      }
+    } else {
+      // Something went wrong in parsing. The parser will print a message. Just add to it.
+      std::cout << "Cannot continue.\n";
+      return 1;
+    }
+  }
+
+  // Print unexpected input arguments
+  if (optind < argc) {
+    printf("Unexpected argument: ");
+    while (optind < argc)
+      printf("%s ", argv[optind++]);
+    printf("\n");
+
+    return 1;
+  }
+
+  // See if the user asked for help
+  bool print_help_and_exit = (parsed_options.find("help") != parsed_options.end());
+  
+  // It is safe to access non-existent values from a map, the result
+  // will be an empty string
+  opt.model = parsed_options["model"];
+  if (opt.model == "") 
+    print_help_and_exit = true;
 
-  if (argc != 2) {
-    std::cerr << "Usage: " << argv[0] << " <model file>" << std::endl;
+  // Enforce that the sample rate is an integer
+  double sample_rate_double = atof(parsed_options["sample-rate"].c_str());
+  if (sample_rate_double != round(sample_rate_double)) {
+    std::cout << "The value of --sample-rate must be an integer.\n";
+    print_help_and_exit = true;
+  }
+  
+  // Print the help and exit
+  if (print_help_and_exit) {
+    std::cout << "Usage: " << argv[0] << " --model <model file> [other options]"
+              << "\nSee the documentation for more information.\n";
     return 1;
   }
 
+  // Collect all other option values. If not set the values will default to 0.
+  opt.output_model_state = parsed_options["output-model-state"];
+  opt.sample_rate        = sample_rate_double;
+  opt.subpixel_offset    = atof(parsed_options["subpixel-offset"].c_str());
+  opt.height_above_datum = atof(parsed_options["height-above-datum"].c_str());
+
+  return 0;
+}
+
+// Read a file's content in a single string
+void readFileInString(std::string const& filename, std::string & str) {
+
+  str.clear(); // clear the output
+
+  std::ifstream ifs(filename.c_str());
+  ifs.seekg(0, std::ios::end);   
+  str.reserve(ifs.tellg());
+  ifs.seekg(0, std::ios::beg);
+  str.assign((std::istreambuf_iterator<char>(ifs)),
+             std::istreambuf_iterator<char>());
+  ifs.close();
+}
+
+// Sort the errors and print some stats
+void printErrors(std::vector<double> & errors) {
+  std::sort(errors.begin(), errors.end());
+
+  if (errors.empty()) {
+    std::cout << "Empty list of errors.\n";
+    return;
+  }
+  
+  std::cout << "Norm of pixel errors after projecting from camera to ground and back.\n";
+  std::cout << "Min:    " << errors[0] << "\n";
+  std::cout << "Median: " << errors[errors.size()/2] << "\n";
+  std::cout << "Max:    " << errors.back() << "\n";
+  std::cout << "Count:  " << errors.size() << "\n";
+}
+
+double pixDiffNorm(csm::ImageCoord const& a, csm::ImageCoord const& b) {
+  return sqrt((a.line - b.line) * (a.line - b.line) + (a.samp - b.samp) * (a.samp - b.samp));
+}
+
+// Load a CSM camera model from an ISD or state file. Return 0 on success.
+int load_csm_camera_model(std::string const& model_file,
```

Comment:
Why return a success/failure code here? If that's the intention, then this should just return a boolean for readability.

Reply:
Fixed in the latest commit.